### PR TITLE
Feature/Refactor: product forms

### LINKED ISSUE #300: [Bug]: Sometimes create new product on container bugs
### What browsers are you seeing the problem on?

Firefox

### Is this happening on the SudoSOS point of sale or on the SudoSOS dashboard?

Dashboard

### Current behaviour

Sometimes when creating a product it fails, and then everything in the modal will slowly break etc. 
image

### Desired behaviour

It works

### Steps to reproduce

Play around with creating a product a few times, at sometimes there will be an error in the console. From that point on selecting options in the modal will not work anymore, and everything will be kinda static. But sometimes nothing breaks and it works like normal

### Other information

_No response_

### LINKED ISSUE #308: [Bug]: Manage products calls the backend too much
### What browsers are you seeing the problem on?

Firefox

### Is this happening on the SudoSOS point of sale or on the SudoSOS dashboard?

Dashboard

### Current behaviour

The frontend calls the same call to the backend for every container:
image

### Desired behaviour

Should be only once, to save load on the backend

### Steps to reproduce

Open network manager from your browser, go to manage products, and enjoy the show

### Other information

_No response_

### PR DESCRIPTION
**Pull request by SuperVK**
**Warning: this pull request should be merged after #229** 

Refactor products forms to use the new form way, increases stability and uniform design, while preserving the reactivity that the product dialogs had. Also introduces a delete button for products.

# Description

image
image
image

Mostly behavior is the same, slight changes is the ability to delete products. In container context they only remove from the container, without a container context they will delete the product, behind a confirm dialog. 

image

Also fixed two bugs (#300 and #308).

## Related issues/external references
#273 (one of many)
Closes #308 
Closes #300 
#263 (one of few)

## Types of changes

- Bug fix _(non-breaking change which fixes an issue)_
- New feature _(non-breaking change which adds functionality)_

### CODE REVIEW

File: apps/dashboard/src/components/container/ContainerActionsDialog.vue [hunk 1]
```diff
@@ -142,20 +147,36 @@ const handleSubmitContainer = handleSubmit(async (values) => {
...
-      emit('create:container', c);
-      closeDialog();
-      toast.add({
-        severity: 'success',
-        summary: t('successMessages.success'),
-        detail: t('successMessages.createContainer'),
-        life: 3000,
+    ).catch((err) => handleError(err, toast));
+
+    if(!container) return;
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/components/container/ContainerActionsDialog.vue]: When can this be the case? And why shouldn't it show some kind of error message?

**Review by CodeNamedRobin — COMMENTED** [review]
Please rebase this first, as now reviews are done on things that are already merged to develop.

**Comment by SuperVK** [comment]
Rebased and should be good now!

File: apps/dashboard/src/components/container/ContainerActionsDialog.vue (hunk 1 above)
Comment (SuperVK) [reply apps/dashboard/src/components/container/ContainerActionsDialog.vue]: Outdated cause from the other PR

File: apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue:1]: The owner does not show up in the dropdown, but saving does accept the request.
image

Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue:1]: The existing price does not format to two decimals (although this is kinda nitpicky)
image

Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue:1]: Might also be nice to have feedback after editing a product, toast would do.

Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue:1]: On mobile the owner part looks kinda funky.
image

File: apps/dashboard/src/components/InputDropdownSpan.vue
Comment (CodeNamedRobin) [inline apps/dashboard/src/components/InputDropdownSpan.vue:1]: Do we want this to be as a separate component or as a v-if-else clause in the InputSpan.vue component?

File: apps/dashboard/src/utils/validation-schema.ts [hunk 2]
```diff
@@ -79,6 +83,19 @@ export const createPayoutSchema =
...
   });
 
+export const createProductSchema =
+    yup.object({
+        name: yup.string().required(),
+        priceInclVat: yup.number().required().default(0),
+        vat: yup.mixed<BaseVatGroupResponse>().required(),
+        category: yup.mixed<ProductCategoryResponse>().required(),
+        alcoholPercentage: yup.number().required().default(0),
+        featured: yup.boolean().required().default(false),
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/utils/validation-schema.ts]: featured, preferred and priceList are not required  by the backend.

**Review by CodeNamedRobin — CHANGES_REQUESTED** [review]
Some nitpicks, some things to think about, some swagger mismatches.

File: apps/dashboard/src/components/InputDropdownSpan.vue
Comment (SuperVK) [reply apps/dashboard/src/components/InputDropdownSpan.vue:1]: I would say this is sufficiently different. Making sure this works in the other needs a lot of extra code that makes it a lot less readable. Cause you need extra options, you need the optionLabel (as extra on top of label). etc

File: apps/dashboard/src/utils/validation-schema.ts (hunk 2 above)
Comment (SuperVK) [reply apps/dashboard/src/utils/validation-schema.ts]: Hahahaha fair point!

File: apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue
Comment (SuperVK) [reply apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue:1]: Yess, good point!

Comment (SuperVK) [reply apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue:1]: Yess, fixed!

Comment (SuperVK) [reply apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue:1]: Yess, this was funky due to a rebase I think

Comment (SuperVK) [reply apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue:1]: Fixed!

Comment (SuperVK) [reply apps/dashboard/src/modules/seller/components/ProductActionExistingProductForm.vue:1]: also fixed btw

**Review by CodeNamedRobin — APPROVED** [review]
LGTM!

**Pull request merged**